src/swsssdk/dbconnector.py

```diff
@@ -27,85 +78,129 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
                 logger.warning(msg)
                 sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
             with open(sonic_db_file_path, "r") as read_file:
-                SonicDBConfig._sonic_db_config = json.load(read_file)
+                # The database_config.json is loaded into the '' index, which referes to the local namespace.
+                SonicDBConfig._sonic_db_config[''] = json.load(read_file)
         except (OSError, IOError):
             msg = "Could not open sonic database config file '{}'".format(sonic_db_file_path)
             logger.exception(msg)
             raise RuntimeError(msg)
         SonicDBConfig._sonic_db_config_init = True
 
     @staticmethod
-    def db_name_validation(db_name):
+    def namespace_validation(namespace):
+        # Load global config if the namespace is an external one.
+        if namespace != '' and SonicDBConfig._sonic_db_global_config_init == False:
+            SonicDBConfig.load_sonic_global_db_config()
```

Comment:
one more thing: namespace_validation() is a library API, if user use namespace_validation() to check namaspace stuff before any load_sonic_db_config, it calls load_sonic_global_db_config() which actually load ns='' case, but didn't set _sonic_db_config_init = TURE, which will make following lines SonicDBConfig.load_sonic_db_config() still load ns='' case again, better to set _sonic_db_config_init = TURE in load_sonic_global_db_config() when ns is ' '.

Reply:
This is a great catch! In stead of make it smarter, how about making it more stupid and simple?

Since use are required to call `load_sonic_global_db_config()` at this use case, let's make this function throw runtime exception and educate user to call `load_sonic_global_db_config` first.

Also take Dong's suggestion to ensure `sonic_db_config_init == True` after calling `load_sonic_global_db_config()`


---
In reply to: [408568390](https://github.com/Azure/sonic-py-swsssdk/pull/63#discussion_r408568390) [](ancestors = 408568390)

Reply:
I am not 100% sure of this use case. The idea I had to make these two independent the load_global_config and local_db_config + to keep the existing behaviour of local_db_config intact.

I can set this flag sonic_db_config_init == True in load_global_config for ' ' case, **but then** the user cannot later if needed override the config present in global_config.json for the current namespace, with **a config which he wants later** .. by calling load_sonic_db_config(sonic_db_file_path=**my_config_file**)

Reply:
It should NOT be allowed to make user override existing config. It only should init once to keep consistence among all applications. Othewise some  application want this config , others want other config, that will be complex.

Reply:
I have updated with this flag sonic_db_config_init = True in the function load_sonic_global_db_config(). 

I haven't removed the implicit call to load_sonic_global_db_config() here if namespace is != '', as other wise the applications ( including scripts which uses it like sonic-db-cli, sonic_cfggen . db_migrator etc ) will have to call it explicitly  SonicDBConfig.load_sonic_global_db_config() ?

Reply:
I am taking Qi's suggestion to make the user explicitly call load_sonic_global_db_config(). 

In addition, I am optimizing the logic a bit by adding a new argument "namespace" to this API  load_sonic_global_db_config(). So if the user inputs an argument "namespace" this API will load the json file for that namespace alone .

This will improve performance of  utilitiles like sonic-cfggen/sonic-db-cli etc where we pass the namespace argument.